feat(NIST 800-53): Add IAM Compliance Checks to Pack (#27)
cdklabs/cdk-nag: RULES.md

| [NIST.800.53-EC2CheckSSHRestricted](https://docs.aws.amazon.com/config/latest/developerguide/restricted-ssh.html)                                       | The Security Group allows unrestricted SSH access - (Control IDs: AC-4, SC-7, SC-7(3)).                         | Not allowing ingress (or remote) traffic from 0.0.0.0/0 or ::/0 to port 22 on your resources helps to restrict remote access.                                                                                                                                                                                                                                 |

> | [NIST.800.53-IAMGroupMembershipCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-no-statements-with-admin-access)              | The IAM user does not belong to any group(s) - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6.).                    | AWS Identity and Access Management (IAM) can help you restrict access permissions and authorizations by ensuring IAM users are members of at least one group. Allowing users more privileges than needed to complete a task may violate the principle of least privilege and separation of duties.                                                            |

Reviewer: ```suggestion
| [NIST.800.53-IAMGroupMembership](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-group-membership-check.html)              | The IAM user does not belong to any group(s) - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6.).                    | AWS Identity and Access Management (IAM) can help you restrict access permissions and authorizations by ensuring IAM users are members of at least one group. Allowing users more privileges than needed to complete a task may violate the principle of least privilege and separation of duties.                                                            |
```

> | [NIST.800.53-IAMNoUserNoPolicyCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-no-policies-check)                               | The IAM policy is attached at the user level - (AC-2(j), AC-3, AC-5c, AC-6).                                    | Assigning privileges at the group or the role level helps to reduce opportunity for an identity to receive or retain excessive privileges.                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| [NIST.800.53-IAMNoUserNoPolicy](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-no-policies-check.html)                               | The IAM policy is attached at the user level - (AC-2(j), AC-3, AC-5c, AC-6).                                    | Assigning privileges at the group or the role level helps to reduce opportunity for an identity to receive or retain excessive privileges.                                                                                                                                                                                                                    |
```

> | [NIST.800.53-IAMNoInlinePolicyCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-no-inline-policy-check)                               | The IAM Group, User, or Role contains an inline policy - (Control ID: AC-6).                                    | AWS recommends to use managed policies instead of inline policies. The managed policies allow reusability, versioning and rolling back, and delegating permissions management.                                                                                                                                                                                |

Reviewer: ```suggestion
| [NIST.800.53-IAMNoInlinePolicy](https://docs.aws.amazon.com/config/latest/developerguide/iam-no-inline-policy-check.html)                               | The IAM Group, User, or Role contains an inline policy - (Control ID: AC-6).                                    | AWS recommends to use managed policies instead of inline policies. The managed policies allow reusability, versioning and rolling back, and delegating permissions management.                                                                                                                                                                                |
```

> | [NIST.800.53-IAMPolicyNoStatementsWithAdminAccess](https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-no-statements-with-admin-access) | The IAM policy grants admin access - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6).                               | AWS Identity and Access Management (IAM) can help you incorporate the principles of least privilege and separation of duties with access permissions and authorizations, by ensuring that IAM groups have at least one IAM user. Placing IAM users in groups based on their associated permissions or job function is one way to incorporate least privilege. |

Reviewer: ```suggestion
| [NIST.800.53-IAMPolicyNoStatementsWithAdminAccess](https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-no-statements-with-admin-access.html) | The IAM policy grants admin access - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6).                               | AWS Identity and Access Management (IAM) can help you incorporate the principles of least privilege and separation of duties with access permissions and authorizations, by ensuring that IAM groups have at least one IAM user. Placing IAM users in groups based on their associated permissions or job function is one way to incorporate least privilege. |
```

### Excluded Rules

Rules from the AWS NIST 800-53 Conformance Pack which could not be implemented.

| AWS Config Rule Name                                                                                       | Summary                                                                                                 | Relevant Control IDs |
| ---------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- | -------------------- |
| [ec2-imdsv2-check](https://docs.aws.amazon.com/config/latest/developerguide/ec2-imdsv2-check.html)         | The EC2 instance has IMDSV2 (instance metadata service version 2) enabled.                              | AC-6                 |

> | [iam-user-mfa-enabled](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-mfa-enabled)      | AWS Identity and Access Management users have multi-factor authentication (MFA) enabled.                | IA-2(1)(2)(11)       |

Reviewer: ```suggestion
| [iam-user-mfa-enabled](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-mfa-enabled.html)      | AWS Identity and Access Management users have multi-factor authentication (MFA) enabled.                | IA-2(1)(2)(11)       |
```

> | [iam-password-policy](https://docs.aws.amazon.com/config/latest/developerguide/iam-password-policy)        | The account password policy for IAM users meets the specified requirements indicated in the parameters. | AC-2(1)              |

Reviewer: ```suggestion
| [iam-password-policy](https://docs.aws.amazon.com/config/latest/developerguide/iam-password-policy.html)        | The account password policy for IAM users meets the specified requirements indicated in the parameters. | AC-2(1)              |
```

> | [iam-group-has-users-check](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-mfa-enabled) | IAM groups have at least one IAM user.                                                                  | AC-2(j)              |

Reviewer: ```suggestion
| [iam-group-has-users-check](https://docs.aws.amazon.com/config/latest/developerguide/iam-group-has-users-check.html) | IAM groups have at least one IAM user.                                                                  | AC-2(j)              |
```

| [NIST.800.53-EC2CheckNoPublicIPs](https://docs.aws.amazon.com/config/latest/developerguide/ec2-instance-no-public-ip.html)                       | The EC2 instance is associated with a public IP address - (Control IDs: AC-4, AC-6, AC-21(b), SC-7, SC-7(3)).   | Amazon EC2 instances can contain sensitive information and access control is required for such resources.                                                        |
| [NIST.800.53-EC2CheckSSHRestricted](https://docs.aws.amazon.com/config/latest/developerguide/restricted-ssh.html)                                | The Security Group allows unrestricted SSH access - (Control IDs: AC-4, SC-7, SC-7(3)).                         | Not allowing ingress (or remote) traffic from 0.0.0.0/0 or ::/0 to port 22 on your resources helps to restrict remote access.                                    |

> | [NIST.800.53-IAMGroupMembershipCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-no-statements-with-admin-access)    | The IAM user does not belong to any group(s) - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6.). | AWS Identity and Access Management (IAM) can help you restrict access permissions and authorizations by ensuring IAM users are members of at least one group. Allowing users more privileges than needed to complete a task may violate the principle of least privilege and separation of duties.                                              |
> | [NIST.800.53-IAMNoUserNoPolicyCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-no-policies-check)     | There is an IAM policy attached at the user level (AC-2(j), AC-3, AC-5c, AC-6).                                 | This rule ensures AWS Identity and Access Management (IAM) policies are attached only to groups or roles to control access to systems and assets. Assigning privileges at the group or the role level helps to reduce opportunity for an identity to receive or retain excessive privileges.|
> | [NIST.800.53-IAMNoInlinePolicyCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-no-inline-policy-check)          | There is an inline IAM policy - (Control ID: AC-6).                                      |  This check ensures an AWS Identity and Access Management (IAM) user, IAM role or IAM group does not have an inline policy to control access to systems and assets. AWS recommends to use managed policies instead of inline policies. The managed policies allow reusability, versioning and rolling back, and delegating permissions management.                      |
> | [NIST.800.53-IAMPolicyNoStatementsWithAdminAccess](https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-no-statements-with-admin-access)          | There is an IAM policy that gives administrator access - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6).   | This check ensures that policies are restricted from containing "Effect": "Allow" with "Action": "\*" over "Resource": "\*". Allowing users to have more privileges than needed to complete a task may violate the principle of least privilege and separation of duties. |

Reviewer: ```suggestion
| [NIST.800.53-IAMGroupMembershipCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-no-statements-with-admin-access)              | The IAM user does not belong to any group(s) - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6.).                    | AWS Identity and Access Management (IAM) can help you restrict access permissions and authorizations by ensuring IAM users are members of at least one group. Allowing users more privileges than needed to complete a task may violate the principle of least privilege and separation of duties.                                                            |
| [NIST.800.53-IAMNoUserNoPolicyCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-no-policies-check)                               | The IAM policy is attached at the user level - (AC-2(j), AC-3, AC-5c, AC-6).                                    | Assigning privileges at the group or the role level helps to reduce opportunity for an identity to receive or retain excessive privileges.                                                                                                                                                                                                                    |
| [NIST.800.53-IAMNoInlinePolicyCheck](https://docs.aws.amazon.com/config/latest/developerguide/iam-no-inline-policy-check)                               | The IAM Group, User, or Role contains an inline policy - (Control ID: AC-6).                                    | AWS recommends to use managed policies instead of inline policies. The managed policies allow reusability, versioning and rolling back, and delegating permissions management.                                                                                                                                                                                |
| [NIST.800.53-IAMPolicyNoStatementsWithAdminAccess](https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-no-statements-with-admin-access) | The IAM policy grants admin access - (Control IDs: AC-2(1), AC-2(j), AC-3, AC-6).                               | AWS Identity and Access Management (IAM) can help you incorporate the principles of least privilege and separation of duties with access permissions and authorizations, by ensuring that IAM groups have at least one IAM user. Placing IAM users in groups based on their associated permissions or job function is one way to incorporate least privilege. |
```

| AWS Config Rule Name                                                                               | Summary                                                                    | Relevant Control IDs |
| -------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------- | -------------------- |
| [ec2-imdsv2-check](https://docs.aws.amazon.com/config/latest/developerguide/ec2-imdsv2-check.html) | The EC2 instance has IMDSV2 (instance metadata service version 2) enabled. | AC-6                 |

> | [iam-user-mfa-enabled](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-mfa-enabled) | AWS Identity and Access Management users have multi-factor authentication (MFA) enabled. | IA-2(1)(2)(11)                 |
> | [iam-password-policy](https://docs.aws.amazon.com/config/latest/developerguide/iam-password-policy) | The account password policy for IAM users meets the specified requirements indicated in the parameters.| AC-2(1)                 |
> | [iam-group-has-users-check](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-mfa-enabled) | IAM groups have at least one IAM user. | AC-2(j)                |

Reviewer: ```suggestion
| [iam-user-mfa-enabled](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-mfa-enabled)      | AWS Identity and Access Management users have multi-factor authentication (MFA) enabled.                | IA-2(1)(2)(11)       |
| [iam-password-policy](https://docs.aws.amazon.com/config/latest/developerguide/iam-password-policy)        | The account password policy for IAM users meets the specified requirements indicated in the parameters. | AC-2(1)              |
| [iam-group-has-users-check](https://docs.aws.amazon.com/config/latest/developerguide/iam-user-mfa-enabled) | IAM groups have at least one IAM user.                                                                  | AC-2(j)              |
```
